linera-execution/src/lib.rs

```diff
@@ -435,6 +442,13 @@ pub trait BaseRuntime {
         content_type: String,
         payload: Vec<u8>,
     ) -> Result<Vec<u8>, ExecutionError>;
+
+    /// Panics if the current time at block validation is `>= timestamp`. Note that block
+    /// validation happens at or after the block timestamp, but isn't necessarily the same.
+    ///
+    /// Cannot be used in fast blocks: A block using this call should be proposed by a regular
+    /// owner, not a super owner.
+    fn assert_before(&mut self, timestamp: Timestamp) -> Result<(), ExecutionError>;
```

Comment:
I'm wondering if this should be contract specific. I'm not sure I understand the use case for services :thinking:

Reply:
I agree, it doesn't make much sense. I'm just thinking that maybe people want to reuse code between contract and service, so if in doubt both should support it? Not sure.